android/anc/src/main/java/org/smartregister/fhircore/anc/data/model/PatientModels.kt

```diff
@@ -106,5 +106,37 @@ data class PatientVitalItem(
   var bpds: String = "",
   var bpdsUnit: String = "",
   var pulse: String = "",
-  var pulseUnit: String = ""
-)
+  var pulseUnit: String = "",
+  var bmi: String = "",
+  var bmiUnit: String = ""
+) {
+  fun isWeightAndHeightAreValid(): Boolean {
+    return weight.isNotEmpty() &&
+      height.isNotEmpty() &&
+      weight.toDouble() > 0 &&
+      height.toDouble() > 0
+  }
+
+  fun isWeightAndHeightAreInMetricUnit(): Boolean {
+    return weightUnit.equals(UnitConstants.UNIT_WEIGHT_METRIC, true) &&
+      heightUnit.equals(UnitConstants.UNIT_HEIGHT_METRIC, true)
+  }
+
+  fun isWeightAndHeightAreInUSCUnit(): Boolean {
```

Comment:
is USC an abbreviation? If so, the SC should be lower case, https://github.com/opensrp/fhircore/wiki#naming-conventions